packages/transformers/js/core/src/hoist.rs

```diff
@@ -668,6 +668,17 @@ impl<'a> Fold for Hoist<'a> {
           }
         }
       }
+      Expr::Ident(ident) => {
+        if let Some(Import { specifier, .. }) = self.collect.imports.get(&id!(ident)) {
+          if specifier != "*" {
+            return Expr::Seq(SeqExpr {
+              span: ident.span,
+              exprs: vec![0.into(), Box::new(Expr::Ident(ident.fold_with(self)))],
+            });
+          }
+        }
+        return Expr::Ident(ident.fold_with(self));
+      }
```

Comment:
A cleaner fix (and what I tried to do first) was to move the `fold_ident` function into this match. But `fold_ident` runs not only referenced identifiers (=`Expr::Ident`) but also the names of function/variable declarations.
I'm not sure if it would make sense to split up fold_ident/extract the parts we need and use them here directly instead.